### ISSUE TITLE
Configure .zenmap location and default to XDG_CONFIG_HOME and other XDG directories

### ISSUE THREAD

**Issue by jcrben** [nmap/nmap#590]
My home folder has this `.zenmap` file. This is a bit annoying because I've literally used Zenmap once, and as far as I can tell I didn't do any configuring of it. This is not an uncommon application behavior which is why my home folder is usually cluttered with dotfiles of programs I've hardly used, which make it trickier to find the stuff that I actually care about.

Proposal:

1. Don't save any configuration (i.e., `zenmap.conf` unless the user deviates from the default settings (i.e., configures the software).
2. Data like `recent_scans.txt` is saved to XDG_DATA_HOME if available
3. Don't mix configuration and state: in `zenmap.conf`, I see this which looks a whole lot like state:
```
[window]
x = 0
y = 23
width = 597
height = 650
```

**Comment by jcrben** [comment]
I'm still around, yep. This is somewhat timely - did you notice https://news.ycombinator.com/item?id=19063727?

> Have you seen an app like this before? So that we could use as a ref. I tried searching a little bit, but all for me have a starting configuration that is saved on disk on first run.

I'm not too sure, but one way you can handle this is by handling these configurations in-memory with a mapping of configuration values to default. Might look something like this in Pythonish pseudocode:

```
def getConfig(option):
   return getConfigFromFile(option) or getConfigFromDefault(option)
```

where getConfigFromDefault() just looks like a hashmap of names to defaults.

I dunno about using `$XDG_STATE_HOME` - I'd probably just toss this in `$XDG_DATA_HOME` which is more widely recognized and also not designed for version control.

Let me know if  you need any more feedback!

**Comment by denisfa** [comment]
> I'm still around, yep. This is somewhat timely - did you notice https://news.ycombinator.com/item?id=19063727?

No, I am just browsing the [Arch Wiki about this](https://wiki.archlinux.org/index.php/XDG_Base_Directory) and seeing what's up with each project.

> configurations in-memory with a mapping of configuration values to default

I can see where you want to go here, but at the same time I believe this is not worth it. The window geometry is part of configuration, so It would need to be split from the rest. Opening the app should automatically request the `zenmap.conf` to be saved, at least for now. If so, there is nothing to do about this.

> I dunno about using `$XDG_STATE_HOME` - I'd probably just toss this in `$XDG_DATA_HOME` which is more widely recognized and also not designed for version control.

Same, I just saw the [related debian page](https://wiki.debian.org/XDGBaseDirectorySpecification#state), so wanted to be sure. These kinda of files were always like data to me. I do understand the issue behind it, though.

**Comment by txtsd** [comment]
Any chance this can be looked at again?
